org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -228,6 +243,32 @@ private static String getStringValue(Map<String, Object> configuration, String k
 		return def;
 	}
 
+	private static Map<String, Object> getMapValue(Map<String, Object> configuration, String key, Map<String, Object> def) {
```

Comment:
can we have a unit test that checks decompiler config is properly parsed as a Map?

Reply:
Now that I think of it, incoming maps are flattened via MapFlattener.flatten in JDTLanguageServer.didChangeConfiguration, so that might be a problem eventually. Have you tried the fernflower config with this change?

Reply:
I haven't, but if maps are being flattened I can't see how it would work. Should I switch back to my command-line-arg settings, or what do you suggest?

Reply:
I can think of many ugly ways to do it, but I need to sleep on it. I need to finish some other task today before I can focus on that issue.

Reply:
Ok so I'm gonna change how flattening is done to make it compatible with nested maps. I got a bit if refactoring to do.

Reply:
Excellent.

Reply:
See #370

Reply:
@dgileadi #370 is merged. Can you please rebase against master, see if it works for you?